.github/workflows/rebase-for-snap-cicd.yaml

```diff
@@ -21,16 +23,10 @@ jobs:
         git config --global user.name "GitHub Actions"
 
         git fetch origin main:main
-        git rebase -X theirs main
-        git submodule update --init --recursive
-
-        mv snap snap.tmp
-        mv snap.tmp/bootstrap snap/
-        rm -rf snap.tmp
+        git reset --hard main
 
-        ./scripts/subiquity_integration
+        ./ci/snap-build-setup.sh ${{ matrix.input }}
         
-        COMMIT_MSG="ci(bootstrap): prep for launchpad snap builder"
         git add -A
-        git commit --amend -m "$COMMIT_MSG" --author="GitHub Actions <github-actions@github.com>"
+        git commit --amend --no-edit # keep original author but set committer to github-actions@github.com
```

Comment:
If we're changing the SHA anyways, isn't it better to just create a new commit?

Reply:
We touched on this during standup and both approaches have flaws. If you add a commit then the message show in launchpad isn't helpful (ie all you see is something like `ci: prep for launchpad snap build`). If you modify the HEAD then it's not signed but at least the commit message matches what is in the git history.

I'm happy either way, so let me know if you have a strong opinion either way!

Reply:
In any case, we should introduce meaningful version numbers for the snaps then, as the associated commit hash will never be part of `main`.

Reply:
Ugh so many issues all because we can't simply do `snapcraft -f path-to-thing.yaml`! Lets discuss at standup. I'd like to maintain the sha over version nos.

Reply:
We could just add the last commit message to the new commit. i.e.
```
ci(release): <Last commit message>
```

Reply:
Correct me if I'm off, but is the aim to be able to trace the build artefact back to a specific commit sha on main?

If so, then piling release commits on `ci/name` might work at the start, but over time the branch's history would become long and problematic. For example, after 50 releases you'd have to go quite far back in the launchpad history to find the HEAD of main -- although it would still be possible! If we amend to avoid stacking commits, the commit in `ci/<name>` would be overridden on the next invocation and so tracing back to a specific sha would only be a matter of looking at HEAD^1 in launchpads history.

I think the solution we actually want is to combine a few bits here:

- I like your updated commit message so will adopt that
- I think we can use [external-metadata](https://snapcraft.io/docs/using-external-metadata) to get info from HEAD^1 to properly version the artefact.

WDTY?

Reply:
> If so, then piling release commits on ci/name might work at the start

Yeah, that doesn't sound like a good solution.

> I like your updated commit message so will adopt that

In a new commit right? Maybe we should add the SHA of the previous commit in the commit message too, then we'd be able to track it "properly", or maybe not worth it since it'll always be HEAD~1?

> I think we can use [external-metadata](https://snapcraft.io/docs/using-external-metadata) to get info from HEAD^1 to properly version the artefact.

Sounds good, we should probably start running `melos version` as part of this workflow too and get the version from there.

Reply:
A dedicated release commit message would be great :+1:
Would it be terribly misleading if we used `snapcraftctl set-version <hash of HEAD~>`?
I'm also not opposed to the `melos version` idea :)

Reply:
Yep agree with all of this. I've created https://github.com/canonical/ubuntu-desktop-provision/issues/456 to get that work done if you're happy to punt that challenge for the moment?